src/robot/api/deco.py

```diff
@@ -68,3 +68,43 @@ def decorator(func):
         func.robot_types = types
         return func
     return decorator
+
+
+def library(scope=None, version=None, robot_auto_kws=False):
+    """Decorator to set custom scope and version and enable/disable public
+    methods that will become keywords.
+
+    This decorator creates ``ROBOT_LIBRARY_SCOPE``, ``ROBOT_LIBRARY_VERSION``
+    and ``ROBOT_AUTO_KEYWORDS`` attributes on the decorated class based on
+    the provided arguments. Robot Framework checks them to determine the
+    class' scope, version, and if methods are disabled from becoming
+    keywords.
+
+    Examples::
+
+        @library(scope='TEST SUITE')
+        class LibraryScope:
+            # ...
+
+        @library(scope='GLOBAL', version='1.3.0')
+        class LibraryScopeAndVersion:
+            # ...
+
+        @library(scope='GLOBAL', version='1.3.0', robot_auto_kws=False)
+        class DecoratedClassDisablePublicMethods:
+            def public_method_is_not_keyword():
+                print('This method will not become keyword')
+
+        @library
+        class DecoratedClassDoesNotPublicDecoratedMethods:
+            @keyword
+            def public_method_is_not_keyword():
+                print('This method will not become keyword')
+     """
+
+    def lib_decorator(func):
+        func.ROBOT_LIBRARY_SCOPE = scope
+        func.ROBOT_LIBRARY_VERSION = version
+        func.ROBOT_AUTO_KEYWORDS = robot_auto_kws
+        return func
+    return lib_decorator
```

Comment:
1. This is a class decorator so argument shouldn't be named `func` but `cls`. The decorator itself could just be named `decorator`, not `lib_decorator`.

2. Need to decide does this decorator or attributes set explicitly have higher precedence. I'll add a separate comment about that.

3. This implementation seems to only work if you pass arguments to the decorator like `@library(version='1.1')`, not if you just use `@library`. See `@keyword` for how to handle that. Tests for just using `@library` are needed as well.